server/api/auth_handler.py

```diff
@@ -0,0 +1,168 @@
+from models.user import User
+from flask import Blueprint, request, make_response, jsonify, g, redirect, url_for
+from flask.views import MethodView
+from server import db, flask_bcrypt
+from functools import wraps
+
+auth_blueprint = Blueprint("auth_blueprint", __name__)
+
+
+class RegisterAPI(MethodView):
+    def post(self):
+        post_data = request.get_json()
+        user = User.query.filter_by(
+            email=post_data.get("email")
+        ).first()
+        if not user:
+            try:
+                if len(post_data.get("password")) < 6:
+                    responseObject = {
+                        "status": "fail",
+                        "message": "Password must be at least six characters long."
+                    }
+                    return make_response(jsonify(responseObject)), 401
+                if post_data.get("password") != post_data.get("confirm"):
+                    responseObject = {
+                        "status": "fail",
+                        "message": "Passwords must match."
+                    }
+                    return make_response(jsonify(responseObject)), 401
+                user = User(
+                    email=post_data.get("email"),
+                    password=post_data.get("password")
+                )
+                db.session.add(user)
+                db.session.commit()
+                auth_token = user.encode_auth_token(user.id)
+                g.user = user
+                responseObject = {
+                    "status": "success",
+                    "message": "Successfully registered.",
+                    "auth_token": auth_token.decode()
+                }
+                resp = make_response(jsonify(responseObject))
+                resp.set_cookie("Authentication token", auth_token.decode(), httponly=True)  # Should the auth_token be encoded or decoded?
```

Comment:
The auth_token should be encoded. It will sit in the client's browser and will be passed automatically on any request they make. On those requests, your auth decorator will decode before allowing access to a protected route.